### DISCUSSION TITLE
How to embed audio in GitHub Markdown file preview? (like README.md)

### DISCUSSION THREAD

**Discussion by X-Raym** [community/community#53410] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,

How to embed an audio player for GitHub Mardown file preview (like in README.md page)?

Tricks I found on Stack overflow seems not working.
They only works with click and drag, not via any remote URL.
Idealy a solution with URL from a file on a GitHub repo will be nice.
Here is a [demo](https://raw.githubusercontent.com/JoepVanlier/Audio_Demos/main/Ravager/dry_wet.mp3) link.

Thx !

**Comment by flacle** [comment]
Hi.

Could you provide the SO pages where the tricks are discussed for reference?

**Comment by X-Raym** [comment]
This was two years ago, I dont remember,
But just found this
https://stackoverflow.com/questions/44185716/add-audio-in-github-readme-md
It claims you can somehow embed audio as .mp4 since 2021. I haven't tested though. Simple mp3 support would be better.

**Comment by flacle** [comment]
Thanks, indeed, this seems to work:
```
ffmpeg -i input.wav -c:a aac -b:a 128k -vn output.mp4
```
Then, drag and drop the MP4 into your README.md using the GitHub web editor. GitHub scripts then generate an AWS asset link that you can then cut-and-paste into the following tag in your markdown text:
```
<video src='https://github.com/user-attachments/assets/some_hash_generated_by_github'/>
```
The only annoyance is that the mute attribute is automatically added, so users have to first click the volume button to unmute, then click on play.

**Comment by X-Raym** [comment]
Ok, thanks for the infos, can be a good enough workaround until a simpler audio embed support, and more audio format support

**Comment by slidesharetool** [comment]
Yeah, GitHub Markdown has pretty strict limitations — most HTML tags like <audio> or iframe-based players won't render on the README preview. It's mostly sanitized for security.

One workaround I tried is linking directly to the raw audio file, but it's just a basic download link — no player UI. For sharing presentations or audio-based slide content more interactively, I sometimes use [this site](https://slidesharetool.com/) to make the content more accessible outside GitHub.

It would be great if GitHub supported inline audio playback directly from repo files, though!

**Comment by M-INDN-SEDTA** [comment]
An **alternative workaround** is to use a **video tag with no image (blank or black frame), resized to look like an audio player**, and enable controls.

However, this only works on **GitHub Pages**, not inside a normal `README.md`, because GitHub **strips all HTML tags like `<video>`** from markdown on repo pages.

---

### ✅ Here's the Alternate Approach — via GitHub Pages:

If you're okay using GitHub Pages (like a custom landing page for your project), here's how you can simulate an audio player using a `<video>` tag:

```html
<video controls width="300" height="50">
  <source src="your-audio-file.mp3" type="audio/mpeg">
  Your browser does not support the video tag.
</video>
```

This will:

* Look and act like an audio player
* Keep the size small (like a media bar)
* Work in all modern browsers

You can embed this on a `index.html` file in your GitHub Pages site.

---

### 👎 Why This Doesn’t Work in README.md

GitHub README files **strip all raw HTML that includes multimedia or scripts**. This includes:

* `<video>`
* `<audio>`
* `<iframe>`

So this method won’t work directly in a `README.md` file, but it works **perfectly** on a GitHub Pages-powered HTML page.

---

### 🔗 Final Suggestion

In your README.md, you can do something like this:

```markdown
🎵 [Click here to listen to the audio player](https://yourusername.github.io/yourrepo/audio.html)
```

Then in `audio.html` (hosted via GitHub Pages), include:

```html
<video controls width="300" height="50">
  <source src="your-audio-file.mp3" type="audio/mpeg">
  Your browser does not support the video tag.
</video>
```

---

Let me know if you want a starter `audio.html` file or help enabling GitHub Pages on your repo.

**Comment by jcodes101** [comment]
so usually GitHub Markdown doesn't really support audio players like HTML does, but here are two solutions that should prove true to help you out.

you can link it directly (similar to how you would link a video if you will)
ex. *[Listen to audio sample](https://example.com/path/to/audio.mp3)*

or you could use an HTML audio tag but it only works outside the Github preview, but to do this it has to be an HTTPS link

ex.

```
<audio controls>
  <source src="https://example.com/path/to/audio.mp3" type="audio/mpeg">
  ...
</audio>
```
